src/boot/efi/boot.c

```diff
@@ -812,12 +828,26 @@ static BOOLEAN menu_run(
                 case KEYPRESS(0, 0, 'p'):
                 case KEYPRESS(0, 0, 'P'):
                         print_status(config, loaded_image_path);
-                        refresh = TRUE;
+                        clear = TRUE;
                         break;
 
                 case KEYPRESS(EFI_CONTROL_PRESSED, 0, 'l'):
                 case KEYPRESS(EFI_CONTROL_PRESSED, 0, CHAR_CTRL('l')):
-                        refresh = TRUE;
+                        clear = TRUE;
+                        break;
+
+                case KEYPRESS(0, 0, 'r'):
+                        if ((INT64)config->console_mode_efivar < ST->ConOut->Mode->MaxMode &&
+                            ST->ConOut->Mode->Mode == ST->ConOut->Mode->MaxMode - 1) {
```

Comment:
you don't set the console mode here. I think you should, i.e. save the mode we found when we initialized, and restore to it here (but let config file setting take precedence)

Reply:
The problem is which mode is the default? We'd need to set the mode first to find out what it really is and if it's supported, but that gets kind of silly when the efi var is set. And in the unlikely case of an invalid initial mode we'd need to decide for one anyway.
I'd rather keep this simply: I made it clear the default will apply on next boot.

Reply:
Why not save `ST->ConOut->Mode->Mode` right when initializing and then simply switching back to that when we wrap around? why wouldn't that work?

Reply:
We are telling the user that we reverted to config (or firmware) default, yet we'd need switch to the defined default first to make sure it's valid/concrete, which seems silly to do if we immediately switch to efivar mode after. Note that auto/max are not concrete modes at that moment yet, so we can't really detect if/when to revert.
Also, if that default mode is unsupported (not only QEMU has problems with mode 1), we'd never reach it for reset purpose, complicating things even more.

Reply:
I am not sure I follow? I can't parse this, can you elaborate?

all i proposed here is that once we wrap around at the end we go to the mode that was in place when we initialized. Or are you saying that `ST->ConOut->Mode->Mode` when we initiaize isn't actually set?